### DISCUSSION TITLE
Infinite Pagination

### DISCUSSION THREAD

**Discussion by akraines** [TanStack/table#3342] [Q&A]
I'm am using RT 7 with pagination. 

Is there a plugin, or how would one go about implementing  Infinite scrolling which fetches data as it get to the last row?
I'm using react-window to handle virtualisation. 
Is there are RT7 hook way of approaching this? 
Or do you recommend going with an approach like building it into the react-window component with something like this: https://github.com/bvaughn/react-window-infinite-loader

**Comment by BalasubramaniM** [comment]
You can use Infinite Queries and achieve this. Send onScroll event to table props and use **fetchNextPage** once you reach the near bottom in your scroll function.

https://react-query.tanstack.com/guides/infinite-queries#_top
